test_simplified_configs.py

```diff
@@ -0,0 +1,162 @@
+from typing import List, Optional
+
+from pydantic import BaseModel, ValidationError
+
+from ert3.config import PluginConfigManager
+from ert3.config import ErtBaseModel, ErtPluginField
+
+
+class _StagesConfig(BaseModel):
+    class Config:
+        validate_all = True
+        validate_assignment = True
+        extra = "forbid"
+        allow_mutation = False
+        arbitrary_types_allowed = True
+
+
+class TransformationConfigBase(_StagesConfig):
+    """Common config for all Transformations"""
+
+    location: str
+
+
+class FileTransformationConfig(TransformationConfigBase):
+    mime: str = ""
+
+
+class SummaryTransformationConfig(TransformationConfigBase):
+    smry_keys: Optional[List[str]] = None
+
+
+class DirectoryTransformationConfig(TransformationConfigBase):
+    pass
+
+
+class DummyInstance:
+    """
+    This is dummy instance that just stores the config.
+    This would be the actual Transformations in the real case.
+    """
+
+    def __init__(self, config) -> None:
+        self._config = config
+
+    def __str__(self) -> str:
+        return f"DummyInstance(config={self._config})"
+
+
+def get_configs(pm):
+    """
+    We now need the create configs that use the ErtPluginField dynamically,
+    as we would like to control the schema created based on the plugins we provide.
+    This will allow us to specify a subset of plugins we want to have effect at runtime,
+    such as only using configs from ert in tests.
+    """
+
+    class StageIO(ErtBaseModel, _StagesConfig, plugin_manager=pm):
```

Comment:
I know you deleted your comment @xjules, but I will leave my response since I think it could be useful for others: 

The ErtBaseModel is the class that specifies that it should use the ErtPluginMetaClass. The intention with the ErtBaseModel was to hide this from the user/developer, similar to how pydantic does it with their BaseModel (which specifies that it should use their ModelMetaClass). However, in our case the use of metaclass is not completly hidden since you still need to specify the plugin_manager argument, but I still think it looks nicer with the use of the ErtBaseModel instead of specifying the metaclass directly. We can also make the plugin_manager argument optional and use a default set of plugins (probably all) if it is not added which would hide this completely, but I like the explicit nature of the current implementation.

https://github.com/sondreso/ert/blob/ed20395152a0bde2c226f6f3eb2bf1816d5de89e/ert3/config/_base.py#L44-L45